### PR TITLE
Tidied up tests, constructor and packages

### PR DESCRIPTION
**Pull request by monstrobishi**
#### What kind of PR is this?:

/kind chore

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/salmon-price-functions/src/price_manager.ts [hunk 1]
```diff
@@ -20,9 +20,6 @@ export class PriceManager {
     private readonly config: PriceSourceConfig,
```
Comment (ivan-zynesis) [inline packages/salmon-price-functions/src/price_manager.ts:20]:
```suggestion
    readonly config: PriceSourceConfig,
```
Shall we have it public? I think allow checking config for instantiated PriceManager is useful.

File: packages/salmon-price-functions/src/price_manager.ts [hunk 2]
```diff
@@ -31,6 +28,10 @@ export class PriceManager {
    * @return {AssetPrice[]}
    */
   async fetchAssetPrices (): Promise<AssetPrice[]> {
+    if (this.config.symbols.length === 0) {
```
Comment (ivan-zynesis) [inline packages/salmon-price-functions/src/price_manager.ts:31]: I personally prefer the original implementation, for QOL reason. Move the `throw` here mean extra/different error handling required on every fetchAssetPrices call

```
constructor(config, provider) {
  this.config = [...config] // make it no longer mutable by caller
}
```

Comment (monstrobishi) [reply packages/salmon-price-functions/src/price_manager.ts:31]: Thanks! I considered this but that is a shallow copy so symbols list would still be mutable, I think a deep copy would complicate the implementation

File: packages/salmon-price-functions/src/price_manager.ts (hunk 1 above)
Comment (monstrobishi) [reply packages/salmon-price-functions/src/price_manager.ts:20]: Yes ok

**Review by canonbrother — APPROVED** [review]

**Review by jingyi2811 — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Review by ivan-zynesis — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**